core/coreapi/unixfs.go

```diff
@@ -126,6 +131,23 @@ func (api *UnixfsAPI) Add(ctx context.Context, files files.File, opts ...options
 		fileAdder.SetMfsRoot(mr)
 	}
 
+	// Get storageMax when node is online.
+	if fileAdder.Pin && !n.EnableGC && n.OnlineMode() {
+		storageUsage, err := n.Repo.GetStorageUsage()
+		if err != nil {
+			return nil, err
+		}
+		storPercentage := float64(storageUsage) / float64(n.StorageMax)
```

Comment:
From `gc.go` I'm seeing that `StorageMax` can be left unset by the user and we'll silently assume a 10GB limit which can make this test a little brittle, a user who just doesn't care about any of this (didn't set the GC, didn't set a max capacity for the repo) will hit this error (which will even prevent it from using IPFS) without understanding what's really going on (and how to fix it).